test/BasicNetworkTest.template.ts

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2021 SYMBOL
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { expect } from 'chai';
+import { Nis1NetworkType, SymbolNetworkType } from '../src/core/constants';
+
+type Network = SymbolNetworkType | Nis1NetworkType;
+
+export const BasicNetworkTester = (Network: any, mainnet: Network, testnet: Network): void => {
+    describe('Network', () => {
+        it('can list all netwroks', () => {
+            // Act:
+            const list = Network.list();
+
+            // Assert:
+            expect(list.length).to.equal(2);
+            expect(list.find((l) => l.name === mainnet.name)).not.to.be.undefined;
+            expect(list.find((l) => l.name === testnet.name)).not.to.be.undefined;
+        });
+
+        it('can find a network by name', () => {
+            // Act:
+            const network = Network.findByName(mainnet.name);
+
+            // Assert:
+            expect(network).not.to.be.undefined;
+            expect(network?.name).to.be.equal(mainnet.name);
+            expect(network?.identifier).to.be.equal(mainnet.identifier);
```

Comment:
not sure if you need `?.` since you're checking that it's defined

[i _think_ the unit test runner you're using stops at first failure]